pkg/lokifrontend/frontend/v2/frontend.go

```diff
@@ -259,13 +286,22 @@ func (f *Frontend) QueryResult(ctx context.Context, qrReq *frontendv2pb.QueryRes
 			level.Warn(f.log).Log("msg", "failed to write query result to the response channel", "queryID", qrReq.QueryID, "user", userID)
 		}
 	}
+	// TODO(chaudum): In case the the userIDs do not match, we do not send a
+	// response to the req.response channel.
+	// In that case, the RoundTripGRPC method waits until the request context deadline exceeds.
+	// Only then the function finished and the request is removed from the
+	// requests map.
 
 	return &frontendv2pb.QueryResultResponse{}, nil
 }
 
 // CheckReady determines if the query frontend is ready.  Function parameters/return
 // chosen to match the same method in the ingester
 func (f *Frontend) CheckReady(_ context.Context) error {
+	if s := f.State(); s != services.Running {
+		return fmt.Errorf("%v", s)
```

Comment:
nit
```suggestion
		return fmt.Errorf("Frontend service not running. Current state: %v", s)
```

Reply:
We already prepend that text in the HTTP handler.

https://github.com/grafana/loki/blob/089ec1b05f5ec15a8851d0e8230153e0eeb4dcec/pkg/loki/loki.go#L569-L576